bridge/client/app/_models/sequence.ts

```diff
@@ -76,9 +76,12 @@ export function createSequenceStateInfo(sequence: ISequence, stageName?: string)
   const warning = !faulty && stages.some((s) => s.latestEvaluation?.result === ResultTypes.WARNING);
   const successful = finished && !faulty && !warning && !aborted;
   const pendingApproval = stages.some((s) => pendingApprovalTypes.includes(s.latestEvent?.type as EventTypes));
-  const icon = latestStage?.latestEvent?.type
-    ? EVENT_ICONS[getShortType(latestStage?.latestEvent?.type)] || EVENT_ICONS.default
-    : EVENT_ICONS.default;
+  const icon =
```

Comment:
This statement is already hard to understand - please simplify it (extract into function, replace ternary operators with `if`s) and write unit tests for it.

Reply:
There are also unit tests for this function. Please extend them accordingly